packages/backend-common/src/tokens/ServerTokenManager.test.ts

```diff
@@ -13,73 +13,95 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+import { getVoidLogger } from '../logging/voidLogger';
 import { ConfigReader } from '@backstage/config';
-import { TokenManager } from './types';
 import { ServerTokenManager } from './ServerTokenManager';
+import { Logger } from 'winston';
+import { JWK } from 'jose';
 
 const emptyConfig = new ConfigReader({});
 const configWithSecret = new ConfigReader({
   backend: { auth: { keys: [{ secret: 'a-secret-key' }] } },
 });
+const env = process.env;
+let logger: Logger;
 
 describe('ServerTokenManager', () => {
-  it('should throw if secret in config does not exist', () => {
-    expect(() => ServerTokenManager.fromConfig(emptyConfig)).toThrowError();
+  beforeEach(() => {
+    process.env = { ...env };
+    logger = getVoidLogger();
   });
 
-  describe('getToken', () => {
-    it('should return a token if secret in config exists', async () => {
-      const tokenManager = ServerTokenManager.fromConfig(configWithSecret);
-      expect((await tokenManager.getToken()).token).toBeDefined();
-    });
+  afterEach(() => {
+    process.env = env;
+  });
 
-    it('should return a token string if using a noop TokenManager', async () => {
-      const tokenManager = ServerTokenManager.noop();
+  describe('getToken', () => {
+    it('should return a token', async () => {
+      const tokenManager = ServerTokenManager.default({
+        config: configWithSecret,
+        logger,
+      });
       expect((await tokenManager.getToken()).token).toBeDefined();
     });
   });
 
   describe('authenticate', () => {
```

Comment:
perhaps an additional `it('should throw for server tokens created using a generated secret', ...`?

Reply:
Done.